tests/test_loom_metamodel.py

```diff
@@ -0,0 +1,352 @@
+    Only checks for errors from the Loom system."""
+
+    with tempdir('bayeslite-loom') as loom_store_path:
+        with bayesdb_open(':memory:') as bdb:
+            bayesdb_register_metamodel(bdb,
+                LoomMetamodel(loom_store_path=loom_store_path))
+            bdb.sql_execute('create table t(u, co, b, ca, cy, nu, no)')
+            for x in xrange(10):
+                cat_dict = ['a', 'b', 'c']
+                bdb.sql_execute('''insert into t(u, co, b, ca, cy, nu, no)
+                    values (?, ?, ?, ?, ?, ?, ?)''',
+                    (
+                        cat_dict[bdb._prng.weakrandom_uniform(3)],
+                        bdb._prng.weakrandom_uniform(200),
+                        bdb._prng.weakrandom_uniform(2),
+                        cat_dict[bdb._prng.weakrandom_uniform(3)],
+                        bdb._prng.weakrandom_uniform(1000)/4.0,
+                        bdb._prng.weakrandom_uniform(1000)/4.0 - 100.0,
+                        bdb._prng.weakrandom_uniform(1000)/4.0
+                    ))
+            bdb.execute('''create population p for t(
+                u unboundedcategorical;
+                co counts;
+                b boolean;
+                ca categorical;
+                cy cyclic;
+                nu numerical;
+                no nominal)
+            ''')
+            bdb.execute('create generator g for p using loom')
+            bdb.execute('initialize 1 model for g')
+            bdb.execute('analyze g for 50 iterations wait')
+            bdb.execute('''estimate probability density of
+                nu = 50, u="a" from p''').fetchall()
+            bdb.execute('''simulate u, co, b, ca, cy, nu, no
+                from p limit 1''').fetchall()
+            bdb.execute('drop models from g')
+            bdb.execute('drop generator g')
+            bdb.execute('drop population p')
+            bdb.execute('drop table t')
+
+
+def test_loom_guess_schema_categorical():
+    """Test to make sure that the LoomMetamodel handles the case where
+    the user provides a categorical variable with more than 256 distinct
+    types. In this case, the metamodel code should know to specify the type
+    unboundedcategorical to loom."""
+
+    with tempdir('bayeslite-loom') as loom_store_path:
+        with bayesdb_open(':memory:') as bdb:
+            bayesdb_register_metamodel(bdb,
+                LoomMetamodel(loom_store_path=loom_store_path))
+            bdb.sql_execute('create table t(v)')
+            vals_to_insert = [''.join(random.choice(string.lowercase) for i in range(20)) for j in range(300)]
+            for i in xrange(len(vals_to_insert)):
+                bdb.sql_execute('''insert into t(v) values (?)''', (vals_to_insert[i],))
+            bdb.execute('create population p for t(v categorical)')
+            bdb.execute('create generator g for p using loom')
+            bdb.execute('initialize 1 model for g')
```

Comment:
I thought we decided that this can't actually be checked because loom does the conversion to unbounded categorical on the spot in the case where the user doesn't specify unbounded categorical but it is necessary to have unbounded categorical because there are more than 256 unique categorical types in the data.